### PR TITLE
github actions for marking issues and PRs as stale

### PR DESCRIPTION
**Pull request by cpelley**
Modified version of the configuration for `actions/stale` from `SciTools/iris`
More info on the action here: https://github.com/actions/stale

To be discussed:
- Choosing a suitable number of days before the action will assign the "Stale" label to a issue/PR (60 days right now).
- Choosing a suitable number of days **after** issue/PR being marked as stale before issue/PR is actually closed (30 days right now).
- Any further label exclusions?

Hopefully will work (permissions on improver repo. setup to allow).

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        # Using actions/stale@v5
+      - uses: actions/stale@3cc123766321e9f15a6676375c154ccffb12a358
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+          # Idle number of days before marking issues/prs stale.
+          days-before-stale: 60
+
+          # Idle number of days before closing stale issues/prs.
+          days-before-close: 30
```
Comment (BelligerG) [inline .github/workflows/stale.yml:26]: Ooo closes issues too?

Is this additive? So it'd be after 90 days of no activity?

Comment (cpelley) [reply .github/workflows/stale.yml:26]: Yep

**Review by BelligerG — DISMISSED** [review]
Looks good to me!

**Comment by cpelley** [comment]
Thanks @BelligerG.  We can similarly target `improver_suite` and `paraflow`.

@bayliffe, are you happy with the proposed timescales and label exceptions?

cheers

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    if: "github.repository == 'metoppv/improver'"
+    runs-on: ubuntu-latest
+    steps:
+        # Using actions/stale@v5
+      - uses: actions/stale@3cc123766321e9f15a6676375c154ccffb12a358
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+          # Idle number of days before marking issues/prs stale.
+          days-before-stale: 60
```
Comment (tjtg) [inline .github/workflows/stale.yml:23]: I'd suggest increasing this time period in order to cater for existing usage of issues, such as #1522, #1737 and #1339. For example, SciTools/iris uses 500 days here.

**Review by tjtg — CHANGES_REQUESTED** [review]
This PR has significant implications for team workflows and longer term maintenance. It would be best to arrange a team discussion to decide on appropriate configuration for the automation added here.

File: .github/workflows/stale.yml (hunk 1 above)
Comment (bayliffe) [reply .github/workflows/stale.yml:26]: We often write issues to capture work for the future, even if we don't intend to action it immediately. We also have chains of work that will take more than 3 months to get through, but for which we have written out the skeleton of work in ticket form. Most of this is on the blue team repo, so would not be affected by this, but in principle I am not sure we want to close issues in any way other than manually.

I'm all for the PR closure. I'd be happy with a shorter period before marking them as stale, say 30 days, then closure 30 days after that, but if that's a bit too short, I'm happy with 90 days.

Comment (cpelley) [reply .github/workflows/stale.yml:26]: Thanks @bayliffe

**Comment by cpelley** [comment]
@bayliffe, @tjtg thanks for getting involved in this.
You have both raised valid concerns over dealing with issues.  There are a range of ways in which concerns can be addressed but as @tjtg rightly points out, these have implications on workflow.

I'm going to suggest we limit this to dealing with 'stale' PRs for now (assuming that is not controversial, @tjtg do chime in) and we then discuss the potential of handling issues offline later?

Cheers

**Comment by cpelley** [comment]
> I'm all for the PR closure. I'd be happy with a shorter period before marking them as stale, say 30 days, then closure 30 days after that, but if that's a bit too short, I'm happy with 90 days. - @bayliffe 

I might agree - I was being conservative.
I think we might better off discuss lowering the current threshold when we also discuss the potential for handling issues with BOM and USAF (in future meeting).

Ready for review.

ping @tjtg

**Review by tjtg — COMMENTED** [review]
I'm happy with this updated configuration - no action taken on issues, PRs marked as stale after 60 days and closed after a further 30 days (90 days total).

**Review by tjtg — APPROVED** [review]

**Comment by cpelley** [comment]
@bayliffe / @BelligerG, you still happy to approve? :)

**Review by bayliffe — APPROVED** [review]

**Comment by cpelley** [comment]
Thanks everyone.  Fingers crossed this works :)

**Pull request merged**